WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -611,11 +631,43 @@ extension MediaCoordinator: Uploader {
                 return
             }
 
-            media.forEach() { self.retryMedia($0) }
+            media.forEach() {
+                self.addObserverForDeletedFiles(for: $0)
+                self.retryMedia($0, isAutomaticRetry: true)
+            }
         }
     }
 }
 
+extension MediaCoordinator {
+    // Based on user logs we've collected for users, we've noticed the app sometimes
+    // trying to upload a Media object and failing because the underlying file has disappeared from
+    // `Documents` folder.
+    // We want to collect more data about that, so we're going to log that info to Sentry,
+    // and also delete the `Media` object, since there isn't really a reasonable way to recover from that failure.
+    func addObserverForDeletedFiles(for media: Media) {
+        addObserver({ (media, state) in
+            guard
+                case .failed(let error) = state,
+                let afError = error as? AFError,
+                case .multipartEncodingFailed(let encodingFailure) = afError else {
+                    return
+            }
+
+            switch encodingFailure {
+            case .bodyPartFileNotReachableWithError,
+                 .bodyPartFileNotReachable:
+                CrashLogging.logMessage("Deleting a media object that's failed to upload because of a missing local file.",
```

Comment:
Considering this is an abnormal (as in "shouldn't happen") condition, why not log this with `CrashLogging.logError(...)`.